### PR TITLE
Migrates to gradle

### PR DESCRIPTION
**Pull request by adriancole**
Ready to go. Will update again once we have a release candidate with test jars.

### CODE REVIEW

**Comment by adriancole** [comment]
cc @abesto @rtyler @jamescway

**Comment by jamescway** [comment]
:+1: lgtm, I guess we should make an issue to publish this also.  Maybe once the pattern settles down in a few days we can duplicate to the other openzipkin projects.

**Comment by adriancole** [comment]
Yeah. That was my thoughts too.

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    zipkinVersion = '1.2.1-SNAPSHOT'
+}
+
+dependencies {
+    compile "org.scala-lang:scala-library:${scalaVersion}"
+
+    compile "io.zipkin:zipkin-common:${zipkinVersion}"
+    compile "org.mongodb:casbah_${scalaInterfaceVersion}:2.8.2"
+
+    testCompile "io.zipkin:zipkin-common:${zipkinVersion}:test"
```
Comment (adriancole) [inline build.gradle]: test jar

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+repositories {
+    jcenter()
+    maven { url 'http://repo.typesafe.com/typesafe/releases/' }
+    maven { url 'https://maven.twttr.com/' }
+}
+
+ext {
+    scalaVersion = '2.10.5'
+    scalaInterfaceVersion = '2.10'
+    zipkinVersion = '1.2.1-SNAPSHOT'
```
Comment (adriancole) [inline build.gradle]: snapshot version. note if build fails, I might need to add `http://oss.jfrog.org/repo` as a repo

File: src/test/scala/com/twitter/zipkin/storage/mongodb/MongoDBSpanStoreTest.scala [hunk 3]
```diff
@@ -3,11 +3,7 @@ package com.twitter.zipkin.storage.mongodb
 import java.nio.ByteBuffer
 
 import com.mongodb.casbah.Imports._
-import com.twitter.app.App
-import com.twitter.util.TimeConversions._
-import com.twitter.zipkin.mongodb.MongoDBSpanStoreFactory
-import com.twitter.zipkin.storage.util.SpanStoreValidator
-import com.twitter.zipkin.storage.{IndexedTraceId, SpanStore}
+import com.twitter.zipkin.storage.{SpanStoreSpec, IndexedTraceId}
```
Comment (adriancole) [inline src/test/scala/com/twitter/zipkin/storage/mongodb/MongoDBSpanStoreTest.scala]: just importing `SpanStoreSpec` for now.. to prove it resolves

**Comment by adriancole** [comment]
switched to snapshot cc @abesto! that said, travis isn't building, so not sure if this works.

File: build.gradle (hunk 2 above)
Comment (adriancole) [inline build.gradle]: yep had to

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+
+compileScala.scalaCompileOptions.additionalParameters = ['-feature']
+compileTestScala.scalaCompileOptions.additionalParameters = ['-feature']
+test.maxParallelForks = 1
+
+repositories {
+    jcenter()
+    maven { url 'https://repo.typesafe.com/typesafe/releases/' }
+    maven { url 'https://oss.jfrog.org/repo' } // for zipkin snapshots
```
Comment (adriancole) [inline build.gradle]: @abesto needed to add this, but made sense

Comment (abesto) [inline build.gradle]: Yep, that's the way it's supposed to be. This should be part of the "using SHAPSHOT releases" section in some imaginary documentation, once it starts existing :)

**Pull request merged**